enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,592 @@
+// Valid values are "MatchRepoDigestOrExact", "MatchRepository", "ExactRepository", "RemapIdentity". When omitted, the default value is "MatchRepoDigestOrExact".
+type IdentityMatchPolicy string
+
+type PolicyMatchExactRepository struct {
+	// repository is the reference of the image identity to be matched.
+  // the value should be a repository (by omitting the tag or digest) in a registry implementing the "Docker Registry HTTP API V2". For example, docker.io/library/busybox
+	// +kubebuilder:validation:Required
+	Repository string `json:"repository"`
+}
+
+type PolicyMatchRemapIdentity struct {
+	// prefix is the prefix of the image identity to be matched.
+  // if the image identity matches the specified prefix, that prefix is replaced by the specified “signedPrefix” (otherwise it is used as unchanged and no remapping takes place)
+  // the prefix and signedPrefix values can be either host[:port] values (matching exactly the same host[:port], string), repository namespaces, or repositories (i.e. they must not contain tags/digests), and match as prefixes of the fully expanded form.
+  // For example, docker.io/library/busybox (not busybox) to specify that single repository, or docker.io/library (not an empty string) to specify the parent namespace of docker.io/library/busybox==busybox).
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// signedPrefix is the prefix of the image identity to be matched in the signature.
+  // the format is the same as "prefix".
+	// +kubebuilder:validation:Required
+	SignedPrefix string `json:"signedPrefix"`
+}
+
+const (
+	IdentityMatchPolicyMatchRepoDigestOrExact IdentityMatchPolicy = "MatchRepoDigestOrExact"
+	IdentityMatchPolicyMatchRepository        IdentityMatchPolicy = "MatchRepository"
+	IdentityMatchPolicyExactRepository        IdentityMatchPolicy = "ExactRepository"
+	IdentityMatchPolicyRemapIdentity          IdentityMatchPolicy = "RemapIdentity"
+)
+
+// +k8s:deepcopy-gen=true
+type ImagePolicyStatus struct {
+
+	// policyJSON contains the whole policy applied to the kubernetes namespace
+	// which got written to disk. This includes cluster-wide policies
+	// from the `openshift-config` namespace as well.
+	PolicyJSON string `json:"policyJSON,omitempty"`
+
+	// conditions provide details on the status of this API Resource.
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Update container runtime config controller to watch ImagePolicy
+
+MCO container runtime config controller ImagePolicy syncHandler updates the signature verification configurations:
+- Retrieves all the ImagePolicy instances on the cluster.
+- Rre-validation: parse the images to validate the policy is not for scopes under OCP product repo. Report the error to machine config logs and the `status` field of that ImagePolicy CR.
```

Comment:
Minor point, should we degrade the MCO here? Just an error log in the MCO (MCC in this case) probably won't be very visible

Reply:
I made a mistake; it's actually an info-level log message about an error encountered while adding an OCP product repo image. The CRCC will ignore this error and proceed with the rollout, similar to how it ignores errors when adding an OCP repo to the block list.
I will update and make it clear.